plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/UIUtils.java

```diff
@@ -1804,6 +1802,22 @@ public static <T> T syncExec(RunnableWithResult<T> runnable) {
         }
     }
 
+    /**
+     * Executes a piece of code in the UI thread and retrieves the result of its calculation.
+     *
+     * @param supplier code to execute
+     * @return result of computation
+     * @param <T> result type
+     */
+    @SuppressWarnings("unchecked")
+    public static <T> T syncExec(Supplier<T> supplier) {
+        final Object[] obj = {null};
+        syncExec(() -> {
+            obj[0] = supplier.get();
+        });
```

Comment:
This block seems redundant

```suggestion
        syncExec(() -> obj[0] = supplier.get());
```

Reply:
Actually, no.  `() -> obj[0] = supplier.get()` returns a value, thus making this method fall into infinite recursion.